sporkfish/endgame_tablebases/lila_tablebase.py

```diff
@@ -2,28 +2,37 @@
 from typing import Optional
 
 import chess
-import chess.syzygy
 import requests
 
+from sporkfish.endgame_tablebases.endgame_tablebase import EndgameTablebase
 
-class LilaTablebase:
-    """Class for handling the lila endgame tablebase in chess engines."""
 
-    base_url = "http://tablebase.lichess.ovh/standard?fen="
+class LilaTablebase(EndgameTablebase):
+    """
+    Class for handling the lila endgame tablebase in chess engines.
 
-    def query_bestmove(board_fen: str) -> Optional[chess.Move]:
+    :ivar BASE_URL: The base URL for the lila endgame tablebase service.
+
+    :method __init__() -> None: Initializes the LilaTablebase instance.
+    :method query(board_fen: str) -> Optional[chess.Move]: Queries the tablebase service for the best move given a board position in FEN format.
+    """
+
+    BASE_URL = "http://tablebase.lichess.ovh/standard?fen="
+
+    def __init__(self) -> None:
+        EndgameTablebase.__init__(self)
+
+    def query(self, board_fen: str) -> Optional[chess.Move]:
```

Comment:
Sorry I missed this earlier - it's good practice to keep the signature of the base class and derived class functions the same. In C++ this code wouldn't compile, but somehow Python allows it (because its dynamically typed). See the things that could potentially go wrong here https://stackoverflow.com/questions/25183424/can-a-python-abstract-base-class-enforce-function-signatures! Could we change this to `query(self, board)` pls?

Reply:
ahh okay and then i do board.fen()